### ISSUE TITLE
blockResources plugin is not working

### ISSUE THREAD

**Issue by ben-obringer** [prerender/prerender#473]
When using the `blockResources` plugin, none of the resources are being blocked for me. Other plugins are working fine though.

In an attempt to fix this issue, I tried to block resources with the following solution: https://github.com/cyrus-and/chrome-remote-interface/wiki/Intercept-and-block-requests-by-URL. Oddly, that solution worked for a while, and then stopped working. I can't seem to figure out why unfortunately.

Overall, could there be some issues with the Chrome DevTools Protocol and certain versions of Chrome or other libraries?

prerender version: 5.2.0
Chrome version: 63.0.3239.132 & 65.0.3324.0 (tested on both)

**Comment by ben-obringer** [comment]
Thank you!

**Comment by thoop** [comment]
It looks like `setBlockedURLs` might not work anymore on certain versions of Chrome even though it's still in the docs...

We'll get something released with `setRequestInterceptor` to get this working on all versions of Chrome.

**Comment by thoop** [comment]
Actually... while testing this more I'm seeing it working locally for me in v63 and v65 of Chrome (I didn't have my blockedResources plugin turned on, doh).

But I'm seeing
```
setBlockedURLs: 
   { [Function: handler]
     category: 'command',
     parameters: { urls: [Object] },
     experimental: true,
     description: 'Blocks URLs from loading.' },
```

Which version of Chrome are you using? Which version of Prerender and which version of chrome-remote-interface do you have installed?

**Comment by ben-obringer** [comment]
Haha. Glad to hear that it's working for you. I'm using the following versions:
Chrome: HeadlessChrome/63.0.3239.132
Prerender: 5.2.0
chrome-remote-interface: 0.24.5

**Comment by thoop** [comment]
Ah, it might be the chrome-remote-interface version 0.24.5. Looks like locally I'm on 0.25.4 and it works just fine.

Can you check and see if your chrome-remote-interface has a `lib/protocol.json` file with version:
```
    "version": {
        "major": "1",
        "minor": "2"
    },
```
and `setBlockedURLs` should look like this from that file:
```
                {
                    "name": "setBlockedURLs",
                    "description": "Blocks URLs from loading.",
                    "parameters": [
                        {
                            "name": "urls",
                            "type": "array",
                            "items": {
                                "type": "string"
                            },
                            "description": "URL patterns to block. Wildcards ('*') are allowed."
                        }
                    ],
                    "experimental": true
                },
```

`name` there should be `"urls"`

If so, can you try upgrading to a newer chrome-remote-interface and see if that fixes it? If so, I'll update our package.json to point to newer versions of our dependencies.

**Comment by ben-obringer** [comment]
Pre and post updating, `lib/protocol.json` looked just like that.
When updating, functions' parameters were no longer set to undefined, but the plugin still isn't working.

**Comment by thoop** [comment]
Interesting. Ok we'll make the switch to use the request interceptor since it looks like that's the way to do it going forward. I'll post back here when that's been updated.

**Comment by ben-obringer** [comment]
Sounds good. Thanks

**Comment by ben-obringer** [comment]
So it turns out the plugin was actually working. I confused myself with the way I was using the library. Sorry about wasting your time.

**Comment by thoop** [comment]
Oh great! Yeah I've had a hard time reproducing the issue. @Artgit can you provide more information on what you're seeing since it looks like it might be working correctly?

**Comment by Artgit** [comment]
@thoop  Hi, this is Alex. I faced the following issue: https://stackoverflow.com/questions/48405977/prerender-io-headless-google-chrome-and-google-analytics

**Comment by laurentdebricon** [comment]
+1 
Version 64.0.3282.186 (Build officiel) (64 bits)

**Comment by hmphu** [comment]
Hello, Is there anybody has solution for this?

**Comment by thoop** [comment]
We'll be switching from using `setBlockedURLs` to `setRequestInterceptor` soon which should fix any issues with the current blocking of certain URLs.

**Comment by thoop** [comment]
This should be fixed now in version 5.4.4

Thanks for reporting!
